### ISSUE TITLE
Throttled SpellCheck

### ISSUE THREAD

**Issue by ccorcos** [electron/electron#10734]
* Electron version: 1.8.1
* Operating system: Mac Sierra

`webFrame.setSpellCheckProvider` should check for misspellings only once the user has topped typing. Otherwise I get these big frames in the middle of typing:

![](https://user-images.githubusercontent.com/1794527/31359337-6b2b109c-acfe-11e7-8087-9e8c8d490bf8.png)

Whenever I press space, there's some jank:

![](https://camo.githubusercontent.com/65fe10204a3e20c94edd264c04fe8b322c9f0e22/687474703a2f2f672e7265636f726469742e636f2f7939367a5054704877722e676966)

However, I notices that Atom doesn't have this same problem because the spellchecker doesn't run until you stop typing:

![](https://camo.githubusercontent.com/731ca5ef9d41f6567b02f3759867819d3b522b9c/687474703a2f2f672e7265636f726469742e636f2f7161446c4149316a416f2e676966)

**Comment by paulcbetts** [comment]
(copy-pasta from electron-spellchecker)

tbh, the core issue is that Electron's spellchecker API is a synchronous API - we must immediately return a response, and especially on macOS, NSSpellchecker sometimes randomly decides to be very slow. Making the spellchecker API async would make solving this issue much easier.

**Comment by po5i** [comment]
Using any spellcheker provider I get the same lack of performance.

**Comment by fantasticsid** [comment]
I've seen similar issue. Also electron seems to call spellchecker provider for quite a few more words than the last one appended once I press space. (not sure if it's the entire content, but it's like hundreds/thousands or misspell checks each time if you have a really long paragraph)

**Comment by sofianguy** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron v2.0.x or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons.

Thanks in advance! Your help is appreciated.

**Comment by sofianguy** [comment]
Thank you for your issue!

We haven't gotten a response to our questions in our comment above. With only the information that is currently in the issue, we don't have enough information to take action. I'm going to close this but don't hesitate to reach out if you have or find the answers we need, we'll be happy to reopen the issue.
